specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/storageTaskAssignments.json

```diff
@@ -0,0 +1,840 @@
+        },
+        "interval": {
+          "type": "integer",
+          "minimum": 1,
+          "format": "int32",
+          "description": "Run interval of task execution. This is a required field when ExecutionTrigger.properties.type is 'OnSchedule'; this property should not be present when ExecutionTrigger.properties.type is 'RunOnce'"
+        },
+        "intervalUnit": {
+          "type": "string",
+          "enum": [
+            "days"
+          ],
+          "x-ms-enum": {
+            "name": "IntervalUnit",
+            "modelAsString": false
+          },
+          "description": "Run interval unit of task execution. This is a required field when ExecutionTrigger.properties.type is 'OnSchedule'; this property should not be present when ExecutionTrigger.properties.type is 'RunOnce'"
+        },
+        "endBy": {
+          "type": "string",
+          "description": "When to end task execution. This is a required field when ExecutionTrigger.properties.type is 'OnSchedule'; this property should not be present when ExecutionTrigger.properties.type is 'RunOnce'"
+        },
+        "startOn": {
+          "type": "string",
+          "description": "When to start task execution. This is an optional field when ExecutionTrigger.properties.type is 'RunOnce'; this property should not be present when ExecutionTrigger.properties.type is 'OnSchedule'"
+        }
+      },
+      "description": "The trigger parameters update for the storage task assignment execution"
+    },
+    "ExecutionConstraints": {
+      "type": "object",
+      "properties": {
+        "abortOnFailures": {
+          "type": "integer",
+          "minimum": 0,
+          "format": "int32",
+          "description": "abortOnFailures constraint for storage task assignment execution"
+        }
+      },
+      "description": "Constraints for storage task assignment execution"
+    },
+    "StorageTaskAssignmentReport": {
+      "type": "object",
+      "properties": {
+        "prefix": {
+          "type": "string",
+          "description": "The prefix of the storage task assignment report"
+        }
+      },
+      "required": [
+        "prefix"
+      ],
+      "description": "The storage task assignment report"
+    },
+    "StorageTaskAssignmentUpdateParameters": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/StorageTaskAssignmentUpdateProperties",
+          "description": "Properties of the storage task assignment."
+        }
+      },
+      "description": "Parameters of the storage task assignment update request"
+    },
+    "StorageTaskAssignmentUpdateProperties": {
+      "type": "object",
+      "properties": {
+        "taskId": {
+          "type": "string",
```

Comment:
```suggestion
          "type": "string",
          "format": "arm-id"
```

Reply:
Not needed since this will be readonly

Reply:
Documenting formats still seems like a generally 'good thing' to me in terms of being clearer documentation. 

Anyone who looks at it and sees format is 'arm-id' can feel more confident they know the set of valid values or meaning or usage.